packages/neon-core/src/u/Fixed8.ts

```diff
@@ -28,9 +55,20 @@ export class Fixed8 extends BN {
   }
 
   public toHex(): string {
-    const hexstring = this.times(100000000)
-      .round(0)
-      .toString(16);
+    let hexstring = "";
+    const num = this.times(DECIMALS).round(0);
+
+    hexstring = num.isLessThan(0)
+      ? TOTAL_FIXED8_HEX.plus(num).toString(16) // convert num to two complement
+      : num.toString(16);
+
+    if (hexstring.length > 16) {
+      throw new Error(

```

Comment:
This is interesting. So we are basically blocking the `toHex()` operation if our final result is outside of Fixed8 boundaries. However, if our internal operations exceed the limits but the final result does not overflow, it is fine and the result will be accurate.

If we really wanted to do this, we should be validating that the number is within bounds after every operation (+, - ,*, / ) so this operation is safe. It does not really make sense to throw on an invalid output when the input is valid.

Reply:
I agree with you that it is better to validate the input of the Fixed8 class. I will add check directly in the Constructor. In this way, we will be stricter and the check will be applied to every operation.